ui/checkboxradio.js

```diff
@@ -0,0 +1,298 @@
+						this.label = ancestor.find( labelSelector );
+					}
+				}
+			}
+			if ( !this.label || !this.label.length ) {
+				$.error( "No label found for checkboxradio widget" );
+			}
+		}
+	},
+
+	_readType: function() {
+		var nodeName = this.element[ 0 ].nodeName.toLowerCase();
+		this.type = this.element[ 0 ].type;
+		if ( nodeName !== "input" || !/radio|checkbox/.test( this.type ) ) {
+			$.error( "Can't create checkboxradio on element.nodeName=" + nodeName +
+				" and element.type=" + this.type );
+		}
+	},
+
+	_enhance: function() {
+		var checked = this.element[ 0 ].checked;
+
+		this._setOption( "disabled", this.options.disabled );
+		this._updateIcon( checked );
+		this._addClass( "ui-checkboxradio", "ui-helper-hidden-accessible" );
+		this._addClass( this.label, "ui-checkboxradio-label", "ui-button ui-widget" );
+
+		if ( this.type === "radio" ) {
+			this._addClass( this.label, "ui-checkboxradio-radio-label" );
+		}
+		if ( checked ) {
+			this._addClass( this.label, "ui-checkboxradio-checked", "ui-state-active" );
+			this._addClass( this.icon, null, "ui-state-hover" );
+		}
+		if ( this.options.label && this.options.label !== this.originalLabel ) {
+			this.label.html( this.icon ? this.icon : "" ).append( this.options.label );
+		} else if ( this.originalLabel ) {
+			this.options.label = this.originalLabel;
+		}
+	},
+
+	widget: function() {
+		return this.label;
+	},
+
+	_getRadioGroup: function() {
+		var name = this.element[ 0 ].name,
+			form = this.element[ 0 ].form,
+			radios = $( [] );
+		if ( name ) {
+			name = name.replace( escapeId, "\\$1" );
+			if ( form ) {
+				radios = $( form ).find( "[name='" + name + "']" );
+			} else {
+				radios = this.document.find( "[name='" + name + "']" )
+					.filter( function() {
+						return !this.form;
+					} );
+			}
+		}
+		return radios.not( this.element );
+	},
+
+	_toggleClasses: function() {
+		var checked = this.element[ 0 ].checked;
+		this._toggleClass( this.label, "ui-checkboxradio-checked", "ui-state-active", checked );
+
+		if ( this.options.icon && this.type === "checkbox" ) {
+
+			// We add ui-state-highlight to change the icon color
+			this._toggleClass( this.icon, null, "ui-icon-check ui-state-highlight", checked )
+				._toggleClass( this.icon, null, "ui-icon-blank", !checked );
+		}
+		if ( this.type === "radio" ) {
+			this._getRadioGroup()
+				.each( function() {
+					var instance = $( this ).checkboxradio( "instance" );
+
+					if ( instance ) {
+						instance._removeClass( instance.label,
+							"ui-checkboxradio-checked", "ui-state-active" );
+					}
+				} );
+		}
+	},
+
+	_destroy: function() {
+		if ( this.icon ) {
+			this.icon.remove();
+			this.iconSpace.remove();
+		}
+
+		widgetCount--;
+		if ( widgetCount === 0 ) {
+			this.formElement.off( "reset." + this.widgetFullName, formResetHandler );
+		}
+	},
+
+	_setOption: function( key, value ) {
+		if ( key === "label" && value === null ) {
+			return this._super( key, this.options[ key ] );
```

Comment:
`_setOption()` doesn't return anything, let's split this on separate lines.

Also, this deserves a comment. I don't understand what this accomplishes.